src/main/java/com/yohanmarcus/webshop/controller/admin/AdminUserChangeController.java

```diff
@@ -0,0 +1,59 @@
+package com.yohanmarcus.webshop.controller.admin;
+
+import com.yohanmarcus.webshop.user.domain.User;
+import com.yohanmarcus.webshop.user.domain.UserRole;
+import com.yohanmarcus.webshop.user.service.UserService;
+import lombok.NoArgsConstructor;
+
+import javax.inject.Inject;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.sql.SQLException;
+
+import static com.yohanmarcus.webshop.util.JspDispatcher.WEB_INF_JSP_ADMIN_USER_CHANGE_JSP;
+import static com.yohanmarcus.webshop.util.JspDispatcher.processRequest;
+
+@NoArgsConstructor
+@WebServlet(name = "userChangeServlet", value = "/admin/users/change")
+public class AdminUserChangeController extends HttpServlet {
+
+  @Inject private UserService userService;
+
+  public AdminUserChangeController(UserService userService) {
+    this.userService = userService;
+  }
+
+  @Override
+  public void doGet(HttpServletRequest req, HttpServletResponse resp)
+      throws ServletException, IOException {
+    try {
+      String id = req.getParameter("id");
+      if (id != null) {
+        User user = userService.findById(id);
+        req.setAttribute("user", user);
+      }
+      processRequest(req, resp, WEB_INF_JSP_ADMIN_USER_CHANGE_JSP);
```

Comment:
om `id` blir null här så leder det fortfarande till att användaren kan se sidan. problemet som detta leder till är då att vi som programmerare har det svårt att felsöka. kanske vi kan kasta ett fel och bara säga vad felet är. ett exempel på error meddelande som "parameter id was empty, so the user could not be fetched"